plugins/net_plugin/net_plugin.cpp

```diff
@@ -2182,14 +2182,27 @@ namespace eosio {
               sync_next_expected_num < sync_last_requested_num );
    }
 
+   // called from c's connection strand
    bool sync_manager::is_sync_request_ahead_allowed(block_num_type blk_num) const REQUIRES(sync_mtx) {
       if (blk_num >= sync_last_requested_num) {
          // do not allow to get too far ahead (sync_fetch_span) of chain head
          // use chain head instead of fork head so we do not get too far ahead of applied blocks
          uint32_t head = my_impl->get_chain_head_num();
          if (blk_num < head + sync_fetch_span)
             return true;
+
+         // might be in irreversible mode
```

Comment:
Do you want to check if it is really in LIB mode?

Reply:
I was originally going to let `net_plugin` determine if in `irreversible` mode, but it is not needed. This condition can't be true unless we are in `irreversible` mode. If it somehow was true we would want to also allow syncing ahead.